### PR TITLE
[chore] update test for confmap changes

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Update test to be compatible with confmap changes to unmarshal via confmap first

### CODE REVIEW

**Review by mx-psi — DISMISSED** [review]

**Comment by mx-psi** [comment]
Doesn't look like this worked :thinking:

**Comment by mx-psi** [comment]
> This passes if core applies [open-telemetry/opentelemetry-collector#9862](https://github.com/open-telemetry/opentelemetry-collector/pull/9862).

Oh, okay, so we need to merge the two PRs together

File: cmd/opampsupervisor/go.mod [hunk 1]
```diff
@@ -34,3 +34,5 @@ require (
 	golang.org/x/sys v0.18.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace go.opentelemetry.io/collector/confmap => github.com/atoulme/opentelemetry-collector/confmap v0.0.0-20240503145806-395864958aba
```
Comment (andrzej-stencel) [inline cmd/opampsupervisor/go.mod:38]: Usage of `atoulme` fork is just temporary?

Comment (atoulme) [reply cmd/opampsupervisor/go.mod:38]: yes, just to prove tests pass with the core changes.

Comment (atoulme) [reply cmd/opampsupervisor/go.mod:38]: https://github.com/open-telemetry/opentelemetry-collector/pull/9862#issuecomment-2100320626 see context here.

File: cmd/otelcontribcol/go.mod [hunk 2]
```diff
@@ -4,8 +4,6 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.10
-
```
Comment (crobert-1) [inline cmd/otelcontribcol/go.mod:8]: I believe we don't want to delete this, and I think dropping this causes the `make genotelcontribcol` failure.

Comment (atoulme) [reply cmd/otelcontribcol/go.mod]: no apologies needed, you were right :)

File: .github/workflows/build-and-test.yml [hunk 3]
```diff
@@ -222,6 +222,7 @@ jobs:
       - name: Gen genotelcontribcol
         run: |
           make genotelcontribcol
+          git diff
```
Comment (dmitryax) [inline .github/workflows/build-and-test.yml:225]: debug?

**Review by dmitryax — APPROVED** [review]
Merging to unblock the build. Please remove the `git diff` line as a follow-up if it's not expected

File: .github/workflows/build-and-test.yml (hunk 3 above)
Comment (atoulme) [reply .github/workflows/build-and-test.yml:225]: yes, it's probably not hurting to leave it

Comment (atoulme) [reply .github/workflows/build-and-test.yml:225]: ah ok I'll remove it

Comment (atoulme) [reply .github/workflows/build-and-test.yml:225]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33209

**Pull request merged**